Libs/MRML/Core/vtkMRMLTextNode.cxx

```diff
@@ -0,0 +1,148 @@
+/*=auto=========================================================================
+
+  Portions (c) Copyright 2009 Brigham and Women's Hospital (BWH) All Rights Reserved.
+
+  See Doc/copyright/copyright.txt
+  or http://www.slicer.org/copyright/copyright.txt for details.
+
+  Program:   3D Slicer
+  Module:    $RCSfile: vtkMRMLTextNode.cxx,v $
+  Date:      $Date: 2006/03/19 17:12:29 $
+  Version:   $Revision: 1.3 $
+
+=========================================================================auto=*/
+
+// SlicerOpenIGTLink MRML includes
+#include "vtkMRMLTextNode.h"
+#include "vtkMRMLTextStorageNode.h"
+
+// MRML includes
+#include <vtkMRMLScene.h>
+
+// VTK includes
+#include "vtkObjectFactory.h"
+#include "vtkXMLUtilities.h"
+
+const int LENGTH_BEFORE_STORAGE_NODE = 256;
+
+//----------------------------------------------------------------------------
+vtkMRMLNodeNewMacro(vtkMRMLTextNode);
+
+//-----------------------------------------------------------------------------
+vtkMRMLTextNode::vtkMRMLTextNode()
+  : Text("")
+  , Encoding(VTK_ENCODING_US_ASCII)
+  , ForceStorageNode(false)
+{
+}
+
+//-----------------------------------------------------------------------------
+vtkMRMLTextNode::~vtkMRMLTextNode()
+{
+  this->SetText("");
+}
+
+//----------------------------------------------------------------------------
+void vtkMRMLTextNode::SetText(std::string text, int encoding/*VTK_ENCODING_NONE*/)
```

Comment:
Performance could be slightly better with const std::string &text (which might matter if the string is long and updates are frequent). If you don't see any drawback then use const ref instead.